helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -893,42 +938,176 @@ private RollingUpdateOp rollingUpdateDeploy(final ZooKeeperClient client,
     }
   }
 
+  private RollingUpdateOp rollingUpdateAwaitStopped(final ZooKeeperClient client,
+                                                    final RollingUpdateOpFactory opFactory,
+                                                    final DeploymentGroup deploymentGroup,
+                                                    final String host) {
+    final TaskStatus taskStatus = getTaskStatus(client, host, deploymentGroup.getJobId());
+    final JobId jobId = deploymentGroup.getJobId();
+
+    if (taskStatus == null) {
+      // Agent has not written job status to zookeeper.
+
+      if (getDeployment(host, jobId) == null) {
+        // The job is not listed under /config/hosts. It may have been manually undeployed?
+        // We can't undeploy it if it's not there anymore, so move on.
+        return opFactory.nextTask();
```

Comment:
Undeploying the job == removing it from `/config/hosts/<host>/jobs` (see e.g.the `undeployJob()` mehtod), right?

So isnt' this branch always going to be true? I put a breakpoint on the line after this block, L958, and ran this in the debugger and it was never hit.